[operation] OperationExecutorImpl_BasicTest.test_getRunningOperationCount

```
java.lang.AssertionError: expected:<3> but was:<0>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl_BasicTest$1.run(OperationExecutorImpl_BasicTest.java:69)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:976)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:993)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl_BasicTest.test_getRunningOperationCount(OperationExecutorImpl_BasicTest.java:64)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-freestyle/330/testReport/junit/com.hazelcast.spi.impl.operationexecutor.impl/OperationExecutorImpl_BasicTest/test_getRunningOperationCount/

--- a/hazelcast/src/test/java/com/hazelcast/spi/impl/operationexecutor/impl/OperationExecutorImpl_BasicTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/spi/impl/operationexecutor/impl/OperationExecutorImpl_BasicTest.java
@@ -16,6 +16,7 @@
 
 package com.hazelcast.spi.impl.operationexecutor.impl;
 
+import com.hazelcast.spi.Operation;
 import com.hazelcast.test.AssertTask;
 import com.hazelcast.test.HazelcastSerialClassRunner;
 import com.hazelcast.test.annotation.QuickTest;
@@ -56,10 +57,11 @@ public class OperationExecutorImpl_BasicTest extends OperationExecutorImpl_Abstr
     public void test_getRunningOperationCount() {
         initExecutor();
 
-        executor.execute(new DummyOperation(GENERIC_PARTITION_ID).durationMs(2000));
-        executor.execute(new DummyOperation(GENERIC_PARTITION_ID).durationMs(2000));
+        CountDownLatch completionLatch = new CountDownLatch(1);
 
-        executor.execute(new DummyOperation(0).durationMs(2000));
+        executor.execute(new LongRunningOperation(GENERIC_PARTITION_ID, completionLatch));
+        executor.execute(new LongRunningOperation(GENERIC_PARTITION_ID, completionLatch));
+        executor.execute(new LongRunningOperation(0, completionLatch));
 
         assertTrueEventually(new AssertTask() {
             @Override
@@ -69,6 +71,22 @@ public class OperationExecutorImpl_BasicTest extends OperationExecutorImpl_Abstr
                 assertEquals(3, runningOperationCount);
             }
         });
+
+        completionLatch.countDown();
+    }
+
+    class LongRunningOperation extends Operation {
+        private CountDownLatch completionLatch;
+
+        public LongRunningOperation(int partitionId, CountDownLatch completionLatch) {
+            this.completionLatch = completionLatch;
+            setPartitionId(partitionId);
+        }
+
+        @Override
+        public void run() throws Exception {
+            completionLatch.await();
+        }
     }
 
     @Test

